statsd/src/main/java/com/avast/metrics/statsd/StatsDMetricsMonitor.java

```diff
@@ -256,4 +256,12 @@ protected String constructMetricName(Optional<String> finalName) {
             return a;
         }
     }
+
+    private String sanitizePrefix(String name) {
+        return name.replaceAll("[^a-zA-Z0-9.]", "_");
+    }
+
+    private String sanitizeNames(String name) {
+        return name.replaceAll("[^a-zA-Z0-9]", "_");
```

Comment:
What's wrong with `-`? It's often used as a separator in kebab-case.

Reply:
I comes from the only documentation we have regarding this: https://docs.datadoghq.com/developers/metrics/#naming-metrics.  Another choice would to to somehow parameterize this.

Reply:
I'm afraid that we also use `-` in metric names...